internal/storage/storage.go

```diff
@@ -349,3 +364,156 @@ func transportConfigS3(config S3Conf) http.RoundTripper {
+	HostKey    string
+}
+
+func newSftpBackend(config SftpConf) (*sftpBackend, error) {
+	// read in and parse pem key
+	key, err := os.ReadFile(config.PemKeyPath)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to read from key file, %v", err)
+	}
+
+	var signer ssh.Signer
+	if config.PemKeyPass == "" {
+		signer, err = ssh.ParsePrivateKey(key)
+	} else {
+		signer, err = ssh.ParsePrivateKeyWithPassphrase(key, []byte(config.PemKeyPass))
+	}
+	if err != nil {
+		return nil, fmt.Errorf("Failed to parse private key, %v", err)
+	}
+
+	// connect
+	conn, err := ssh.Dial("tcp", config.Host+":"+config.Port,
+		&ssh.ClientConfig{
+			User: config.UserName,
+			Auth: []ssh.AuthMethod{
+				ssh.PublicKeys(signer),
+			},
+			HostKeyCallback: TrustedHostKeyCallback(config.HostKey),
+		},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to start ssh connection, %v", err)
+	}
+	// defer conn.Close()
+
+	// create new SFTP client
+	client, err := sftp.NewClient(conn)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to start sftp client, %v", err)
+	}
+	// defer client.Close()
+
+	sfb := &sftpBackend{
+		Connection: conn,
+		Client:     client,
+		Conf:       &config,
+	}
+
+	_, err = client.ReadDir("./")
+
+	if err != nil {
+		return nil, fmt.Errorf("Failed to list files with sftp, %v", err)
+	}
+
+	return sfb, nil
+}
+
+// NewFileWriter returns an io.Writer instance for the sftp remote
+func (sfb *sftpBackend) NewFileWriter(filePath string) (io.WriteCloser, error) {
+	if sfb == nil {
+		return nil, fmt.Errorf("Invalid sftpBackend")
+	}
+	// Make remote directories
+	parent := filepath.Dir(filePath)
+	err := sfb.Client.MkdirAll(parent)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create dir with sftp, %v", err)
+	}
+
+	file, err := sfb.Client.OpenFile(filePath, os.O_CREATE|os.O_TRUNC|os.O_RDWR)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create file with sftp, %v", err)
+	}
+
+	return file, nil
+}
+
+// GetFileSize returns the size of the file
+func (sfb *sftpBackend) GetFileSize(filePath string) (int64, error) {
+	if sfb == nil {
+		return 0, fmt.Errorf("Invalid sftpBackend")
+	}
+
+	stat, err := sfb.Client.Lstat(filePath)
+	if err != nil {
+		return 0, fmt.Errorf("Failed to get file size with sftp, %v", err)
+	}
+
+	return stat.Size(), nil
+}
+
+// NewFileReader returns an io.Reader instance
+func (sfb *sftpBackend) NewFileReader(filePath string) (io.ReadCloser, error) {
+	if sfb == nil {
+		return nil, fmt.Errorf("Invalid sftpBackend")
+	}
+
+	file, err := sfb.Client.Open(filePath)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to open file with sftp, %v", err)
+	}
+
+	return file, nil
+}
+
+// RemoveFile removes a file or an empty directory.
+func (sfb *sftpBackend) RemoveFile(filePath string) error {
+	if sfb == nil {
+		return fmt.Errorf("Invalid sftpBackend")
+	}
+
+	err := sfb.Client.Remove(filePath)
+	if err != nil {
+		return fmt.Errorf("Failed to remove file with sftp, %v", err)
+	}
+
+	return nil
+}
```

Comment:
These are not required for sync functionality

Reply:
Indeed, I just thought that instead of just adding placeholder functions for the rest of the methods implemented by the Backend interface, I might as well make these functional implementations, for completeness.